src/server/table_hotspot_policy.cpp

```diff
@@ -5,36 +5,114 @@
+        _points[i].read_hotpartition_counter.init_app_counter("app.pegasus",
+                                                              read_counter_name.c_str(),
+                                                              COUNTER_TYPE_NUMBER,
+                                                              read_counter_desc.c_str());
+        string write_paritition_desc = _app_name + '.' + "write." + std::to_string(i);
+        write_counter_name = fmt::format("app.stat.hotspots@{}", write_paritition_desc);
+        write_counter_desc = fmt::format("statistic the hotspots of app {}", write_paritition_desc);
+        _points[i].write_hotpartition_counter.init_app_counter("app.pegasus",
+                                                               write_counter_name.c_str(),
+                                                               COUNTER_TYPE_NUMBER,
+                                                               write_counter_desc.c_str());
+    }
+}
+
+/*static*/ void hotspot_calculator::notice_replica(const std::string &app_name,
+                                                   const int partition_index,
+                                                   const bool if_read_request)
+{
+    ddebug("start to notice_replica, %s.%d %s",
+           app_name.c_str(),
+           partition_index,
+           if_read_request ? "read_data" : "read_datawrite_data");
+    std::vector<rpc_address> meta_servers;
+    replica_helper::load_meta_servers(meta_servers);
+    rpc_address meta_server;
+
+    meta_server.assign_group("meta-servers");
+    for (auto &ms : meta_servers) {
+        meta_server.group_address()->add(ms);
     }
+    auto cluster_name = replication::get_current_cluster_name();
+    auto resolver = partition_resolver::get_resolver(cluster_name, meta_servers, app_name.c_str());
+    ::dsn::apps::start_hotkey_detect_request req;
+    req.type = if_read_request ? dsn::apps::hotkey_type::READ : dsn::apps::hotkey_type::WRITE;
+    resolver->call_op(RPC_START_DETECT_HOTKEY,
+                      req,
+                      nullptr,
+                      [app_name, partition_index, if_read_request](
+                          error_code err, dsn::message_ex *request, dsn::message_ex *resp) {
+                          if (err == ERR_OK) {
+                              ::dsn::apps::start_hotkey_detect_response response;
+                              ::dsn::unmarshall(resp, response);
+                              if (response.err == ERR_OK) {
+                                  ddebug("detect hotspot rpc sending succeed");
+                                  return;
+                              } else if (response.err == ERR_SERVICE_ALREADY_EXIST) {
+                                  ddebug("this hotspot rpc has been sending");
+                              }
+                          } else if (err == ERR_TIMEOUT) {
+                              notice_replica(app_name, partition_index, if_read_request);
```

Comment:
If `ERR_TIMEOUT`, the policy will keep retrying forever？if this, I think it shoud be a `max_retry_time`